plugins/module_utils/WebRequest.psm1

```diff
@@ -199,14 +199,17 @@ Function Get-AnsibleWindowsWebRequest {
     }
 
     # Enable TLS1.1/TLS1.2 if they're available but disabled (eg. .NET 4.5)
-    $security_protocols = [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::SystemDefault
-    if ([System.Net.SecurityProtocolType].GetMember("Tls11").Count -gt 0) {
-        $security_protocols = $security_protocols -bor [System.Net.SecurityProtocolType]::Tls11
-    }
-    if ([System.Net.SecurityProtocolType].GetMember("Tls12").Count -gt 0) {
-        $security_protocols = $security_protocols -bor [System.Net.SecurityProtocolType]::Tls12
+    $NetFrameworkVersion = (Get-ItemProperty -LiteralPath "HKLM:SOFTWARE\Microsoft\NET Framework Setup\NDP\v4\Full").Release
+    if ($NetFrameworkVersion -lt "528449") {
+        $security_protocols = [System.Net.ServicePointManager]::SecurityProtocol -bor [System.Net.SecurityProtocolType]::SystemDefault
+        if ([System.Net.SecurityProtocolType].GetMember("Tls11").Count -gt 0) {
+            $security_protocols = $security_protocols -bor [System.Net.SecurityProtocolType]::Tls11
+        }
+        if ([System.Net.SecurityProtocolType].GetMember("Tls12").Count -gt 0) {
+            $security_protocols = $security_protocols -bor [System.Net.SecurityProtocolType]::Tls12
+        }
+        [System.Net.ServicePointManager]::SecurityProtocol = $security_protocols
     }
```

Comment:
```suggestion
    # This mess is required to enable TLS 1.1, 1.2 on versions older than
    # .NET Framework 4.7. It also forces a PSRemoting PowerShell host to use
    # the OS TLS protocol policies rather than a hardcoded set of TLS 1.0,
    # 1.1, and 1.2 only.
    # https://github.com/ansible-collections/ansible.windows/pull/573#issuecomment-1865477707
    $security_protocols = [System.Net.ServicePointManager]::SecurityProtocol
    if ([System.Net.SecurityProtocolType].GetMember("Tls13")) {
        # If the Tls13 member is present we are on .NET Framework 4.8 so using
        # the SystemDefault setting will use the OS policies. If it's not set
        # to SystemDefault already we are running in a PSRemoting WSMan host
        # and need some reflection to reconfigure the policies to allow this.

        if ($security_protocols -ne 'SystemDefault') {
            # https://learn.microsoft.com/en-us/dotnet/framework/network-programming/tls#switchsystemnetdontenablesystemdefaulttlsversions
            $disableSystemTlsField = [System.Net.ServicePointManager].GetField(
                's_disableSystemDefaultTlsVersions',
                [System.Reflection.BindingFlags]'NonPublic, Static')
            if ($disableSystemTlsField -and $disableSystemTlsField.GetValue($null)) {
                $disableSystemTlsField.SetValue($null, $false)
            }

            [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.SecurityProtocolType]::SystemDefault
        }
    }
    else {
        # We are on .NET 4.7 or older, as TLS 1.2 is the max version we can
        # use here we just manually enable the protocols known to the runtime.
        if ([System.Net.SecurityProtocolType].GetMember("Tls11")) {
            $security_protocols = $security_protocols -bor [System.Net.SecurityProtocolType]::Tls11
        }
        if ([System.Net.SecurityProtocolType].GetMember("Tls12")) {
            $security_protocols = $security_protocols -bor [System.Net.SecurityProtocolType]::Tls12
        }
        [System.Net.ServicePointManager]::SecurityProtocol = $security_protocols
    }
```

It's better to use a runtime check here rather than seeing what version is installed in the registry. This allows us to eventually run on PowerShell 7+ and use its sane defaults.

Essentially this suggested code will ensure that running on .NET Framework 4.8 or newer will always use the OS policies when it comes to selecting the protocols. Older framework versions will stick to just TLS 1.1 and 1.2 but that's ok because they'll never support a newer version anyway.